src/redis-cli.c

```diff
@@ -2403,6 +2414,8 @@ static int issueCommandRepeat(int argc, char **argv, long repeat) {
         }
         if (cliSendCommand(argc,argv,repeat) != REDIS_OK) {
             cliPrintContextError();
+            redisFree(context);
+            context = NULL;
```

Comment:
was this a plain leak? found it with valgrind?

Reply:
actually this is not a leak (or a harmless one), we will free it in the next call.
i find it in https://github.com/redis/redis/pull/10382/commits/3d0b7f82b3d7b7a700ab060ef9ee91b684118e78
(not sure i do it right, but it look like no harm (added the free))
```
        if (config.cluster_reissue_command || context == NULL ||
            context->err == REDIS_ERR_IO || context->err == REDIS_ERR_EOF)         -->>
        {
            if (cliConnect(CC_FORCE) != REDIS_OK) {            ---- here
                cliPrintContextError();
                config.cluster_reissue_command = 0;
                return REDIS_ERR;
            }
        }

static int cliConnect(int flags) {
    if (context == NULL || flags & CC_FORCE) {
        if (context != NULL) {
            redisFree(context);
            config.dbnum = 0;
            config.in_multi = 0;
            cliRefreshPrompt();
        }
```